controlplane/kubeadm/api/v1alpha3/kubeadm_control_plane_webhook.go

```diff
@@ -384,6 +386,52 @@ func (in *KubeadmControlPlane) validateEtcd(prev *KubeadmControlPlane) (allErrs
 	return allErrs
 }
 
+func (in *KubeadmControlPlane) validateVersion(previousVersion string) (allErrs field.ErrorList) {
+	fromVersion, err := util.ParseMajorMinorPatch(previousVersion)
+	if err != nil {
+		allErrs = append(allErrs,
+			field.InternalError(
+				field.NewPath("spec", "version"),
+				errors.Wrapf(err, "failed to parse current kubeadmcontrolplane version: %s", previousVersion),
+				),
+		)
+		return allErrs
+	}
+
+	toVersion, err := util.ParseMajorMinorPatch(in.Spec.Version)
+	if err != nil {
+		allErrs = append(allErrs,
+			field.InternalError(
+				field.NewPath("spec", "version"),
+				errors.Wrapf(err, "failed to parse current kubeadmcontrolplane version: %s", previousVersion),
+			),
+		)
+		return allErrs
+	}
+
+	if toVersion.LT(fromVersion) {
+		allErrs = append(allErrs,
+			field.Forbidden(
+				field.NewPath("spec", "version"),
+				fmt.Sprintf("cannot downgrade control plane from %s to %s", previousVersion, in.Spec.Version),
+			),
+		)
+	} else {
+		if toVersion.Major - fromVersion.Major > 1 ||
+			(toVersion.Major - fromVersion.Major == 1 && toVersion.Minor != 0) ||
+			(toVersion.Major == fromVersion.Major && toVersion.Minor - fromVersion.Minor > 1) {
```

Comment:
Can we add comment around this? It's a little hard to follow what we're doing here.

In alternative to comparing one by one, we could create a "ceiling version" from the information we have in the `fromVersion`, and then make sure the `toVersion` is less or equal to the `ceilingVersion`, wdyt?

Reply:
Defining a ceiling version might be harder than expected. Mainly because, say for instance, the current KCP version is `1.17.4`. Then, we could decide that the ceiling version is `1.18.x`.
But it could also be the case, that the next version actually released was a major version bump, say `2.0.0`.  So, (to tackle this scenario), if we keep the ceiling as `2.0.0`, then ideally, an update could skip over multiple minor versions which is what we wanna avoid.
Does it make sense?

Reply:
Yeah, although the plan for now is to stay on Kubernetes 1.x, if we ever go down the route of a major release, there is lots of changes that need to happen

Reply:
Honestly I could remove it and keep the logic simple enough.